pipeline/configuration.py

```diff
@@ -0,0 +1,59 @@
+import base64
+import json
+import os
+import sys
+from pathlib import Path
+from platform import python_version
+from typing import TypedDict
+
+from packaging import version
+
+from pipeline.util.logging import _print
+
+PIPELINE_CACHE = os.getenv(
+    "PIPELINE_CACHE",
+    Path(os.getenv("LOCALAPPDATA")) / ".pipeline/"
+    if (sys.platform == "win32" or sys.platform == "cygwin")
+    else Path.home() / ".cache/pipeline/",
+)
+
+if isinstance(PIPELINE_CACHE, str):
+    PIPELINE_CACHE = Path(PIPELINE_CACHE)
```

Comment:
`pathlib.Path` will happily eat a `pathlib.Path`!

```suggestion
PIPELINE_CACHE = Path(os.getenv(
    "PIPELINE_CACHE",
    Path(os.getenv("LOCALAPPDATA")) / ".pipeline/"
    if (sys.platform == "win32" or sys.platform == "cygwin")
    else Path.home() / ".cache/pipeline/",
))
```